openedx/core/djangoapps/programs/management/commands/backpopulate_program_credentials.py

```diff
@@ -0,0 +1,111 @@
+from collections import namedtuple
+import logging
+
+from django.core.management import BaseCommand, CommandError
+from django.db.models import Q
+from opaque_keys.edx.keys import CourseKey
+from provider.oauth2.models import Client
+
+from certificates.models import GeneratedCertificate
+from openedx.core.djangoapps.programs.models import ProgramsApiConfig
+from openedx.core.djangoapps.programs.tasks.v1.tasks import award_program_certificates
+from openedx.core.djangoapps.programs.utils import get_programs
+
+
+# TODO: Log to console, even with debug mode disabled?
+logger = logging.getLogger(__name__)
+RunMode = namedtuple('RunMode', ['course_key', 'mode_slug'])
+
+
+class Command(BaseCommand):
+    """Management command for backpopulating missing program credentials.
+
+    The command's goal is to pass a narrow subset of usernames to an idempotent
+    Celery task for further (parallelized) processing.
+    """
+    help = 'Backpopulate missing program credentials.'
+    client = None
+    run_modes = None
```

Comment:
@schenedx so that I can use helper methods to break up the code and make it more readable without passing arguments around unnecessarily.